syntax = "proto3";

service AuthService {
  rpc Authenticate(AuthRequest) returns (AuthResponse);
  rpc RefreshToken(TokenRequest) returns (AuthResponse);
  rpc VerifyToken(TokenRequest) returns (TokenResponse);
}

message AuthRequest {
  string email = 1;
  string password = 2;
}

message TokenRequest {
  string token = 1;
}

message AuthResponse {
  bool success = 1;
  string message = 2;
  string access_token = 3;
  string refresh_token = 4;
}

message TokenResponse {
  bool valid = 1;
  string message = 2;
}
